http-server-netty/src/main/java/io/micronaut/http/server/netty/NettyHttpRequest.java

```diff
@@ -200,34 +197,6 @@ public NettyHttpRequest(io.netty.handler.codec.http.HttpRequest nettyRequest,
         this.channelHandlerContext = ctx;
         this.headers = new NettyHttpHeaders(nettyRequest.headers(), conversionService);
         this.body = ByteBody.of(nettyRequest);
-        this.contentLength = headers.contentLength().orElse(-1);
```

Comment:
are these three actually worthwhile? i would guess they are always accessed anyway.

Reply:
After thinking about it more, I will move the caching to headers and avoid nullifying it. Keep it this way for now. Routing should avoid checking the headers in cases when it will always match for any content-type/accepts.

Reply:
i think always matching is rare, it's not the default